src/stream/src/executor/mview/materialize.rs

```diff
@@ -545,23 +569,53 @@ impl<SD: ValueRowSerde> MaterializeCache<SD> {
         let mut buffered = stream::iter(futures).buffer_unordered(10).fuse();
         while let Some(result) = buffered.next().await {
             let (key, value) = result;
-            self.data.push(key, value?);
+            match conflict_behavior {
+                ConflictBehavior::Overwrite => self.data.push(key, CacheValue::Overwrite(value?)),
+                ConflictBehavior::IgnoreConflict => match value? {
+                    // for ignore conflict, we only need to check whether the record exist in cache,
+                    // and do not need it's value.
+                    Some(_) => self.data.push(key, CacheValue::Ignore(Some(()))),
+                    None => self.data.push(key, CacheValue::Ignore(None)),
+                },
+                _ => unreachable!(),
+            };
         }
 
         Ok(())
     }
 
-    pub fn force_get(&mut self, key: &[u8]) -> &Option<CompactedRow> {
-        self.data.get(key).unwrap_or_else(|| {
+    pub fn force_get_for_overwrite(&mut self, key: &[u8]) -> &Option<CompactedRow> {
+        if let CacheValue::Overwrite(cache_row) = self.data.get(key).unwrap_or_else(|| {
+            panic!(
+                "the key {:?} has not been fetched in the materialize executor's cache ",
+                key
+            )
+        }) {
+            cache_row
+        } else {
+            unreachable!()
+        }
+    }
+
+    pub fn force_get_for_ignore(&mut self, key: &[u8]) -> &Option<EmptyValue> {
+        if let CacheValue::Ignore(cache_row) = self.data.get(key).unwrap_or_else(|| {
             panic!(
                 "the key {:?} has not been fetched in the materialize executor's cache ",
                 key
             )
-        })
+        }) {
+            cache_row
+        } else {
+            unreachable!()
+        }
     }
 
     pub fn put(&mut self, key: Vec<u8>, value: Option<CompactedRow>) {
-        self.data.push(key, value);
+        self.data.push(key, CacheValue::Overwrite(value));
+    }
+
+    pub fn put_without_value(&mut self, key: Vec<u8>) {
+        self.data.push(key, CacheValue::Ignore(Some(())));
     }
```

Comment:
Maybe we can write the logic directly in `handle_conflict` instead of defining these two methods.